ui/packages/catalog-ui-search/src/main/webapp/js/model/User.js

```diff
@@ -213,7 +213,7 @@ User.Preferences = Backbone.AssociatedModel.extend({
   },
   savePreferences: function() {
     const currentPrefs = this.toJSON()
-    if (_.isEqual(currentPrefs, this.lastSaved)) {
+    if (_.isEqual(Common.duplicate(currentPrefs), this.lastSaved)) {
```

Comment:
I wonder if it would be equivalent to just stringify in both places instead of duplicate, and then compare the strings.  Not sure if there is a risk of some strings not matching even though the json object it represents matching, but since they are both generated the same way, there should be no weirdness like whitespace.

Might be a more efficient solution as the re-parsing of these objects may be overkill

Reply:
The issue with comparing the JSON strings is key ordering which isn't guaranteed. The `Common.duplicate` is the better solution. I would only duplicate on the save though.

Reply:
We had it that way originally, and it did not work.  I believe duplicate just stored the data, and not other functions on the objects.  So this would return true in cases where not duplicating returned false.